File: tests/fixtures/agent_generated/agent_auth_missing.ts
// Agent-generated: AI assistant creates API routes without auth middleware
// Expected: SEC-002 (auth_bypass_ts)

import express from "express";

const app = express();
app.use(express.json());

const users: Record<string, any> = {};

// No auth middleware applied — all routes publicly accessible
app.get("/api/users", (req, res) => {
  res.json(Object.values(users));
});

app.delete("/api/users/:id", (req, res) => {
  delete users[req.params.id];
  res.json({ status: "deleted" });
});

app.post("/api/admin/config", (req, res) => {
  // Admin config endpoint with no authentication
  res.json({ status: "config updated", config: req.body });
});

app.post("/api/admin/reset", (req, res) => {
  // Dangerous reset operation with no auth
  Object.keys(users).forEach((k) => delete users[k]);
  res.json({ status: "database reset" });
});

File: tests/fixtures/agent_generated/agent_auth_missing.py
# Agent-generated: AI assistant creates admin routes without auth middleware
# Expected: SEC-002 (auth_bypass)

from flask import Flask, jsonify, request

app = Flask(__name__)

users_db = {}

@app.route("/api/users", methods=["GET"])
def list_users():
    """List all users — no authentication required."""
    return jsonify(list(users_db.values()))

@app.route("/api/users/<int:user_id>", methods=["DELETE"])
def delete_user(user_id):
    """Delete a user — no authentication check."""
    if user_id in users_db:
        del users_db[user_id]
        return jsonify({"status": "deleted"})
    return jsonify({"error": "not found"}), 404

@app.route("/api/admin/settings", methods=["POST"])
def update_settings():
    """Update admin settings — no auth, no role check."""
    settings = request.json
    return jsonify({"status": "updated", "settings": settings})

@app.route("/api/admin/reset", methods=["POST"])
def reset_database():
    """Reset database — dangerous operation with no auth."""
    users_db.clear()
    return jsonify({"status": "reset complete"})
